R/weight.R

```diff
@@ -68,6 +89,49 @@ setMethod("weightVariables", "VariableCatalog", function (x) {
     }
 })
 
+#' @rdname weightVariables
+#' @export
+`weightVariables<-` <- function(x, value) {
+    if (is.dataset(x)) {
+        varCat <- allVariables(x)
+    } else if (inherits(x,  "VariableCatalog")) {
+        varCat <- x
+    } else {
+        halt("Left hand side is not a Crunch Dataset or Variable Catalog.")
+    }
+    current <- crGET(shojiURL(varCat, "orders", "weights"))
+    new <- current
+    if (is.null(value)) {
+        new$graph <- NULL
+    } else {
+        if (is.variable(value) || length(value) == 1) {
+            value <- list(value)
+        }
+        var_names <- vapply(value, function(x){
```

Comment:
You don't need this unless the input is invalid, so move this work inside the if at L118